### DISCUSSION TITLE
[git][windows] Preventing local file bloat

### DISCUSSION THREAD

**Discussion by hl2guide** [community/community#23027] [New to GitHub]
<p>I’m using GitHub Desktop and git.exe for my repo and am running into an issue.</p>
<p>Over time the local repos folder bloats out to over <strong>500MB</strong> for a remote repo that is <strong>under</strong>  <strong>11MB</strong>.</p>
<p>I have a PowerShell script that runs the following commands:</p>
<pre><code>clean -f
checkout --orphan TEMP_BRANCH
add -A
commit -am "Initial Commit"
branch -D master
branch -m master
push -f origin master
</code></pre>
<p>I’m trying to get it to:</p>
<ol>
<li>retain the latest commit only</li>
<li>delete all old history and old files</li>
<li>clean the .git folder</li>
</ol>
<p>for both remote and local.</p>
<p>Any ideas?</p>

**Comment by fire-eggs** [comment]
<pre><code>checkout --orphan TEMP\_BRANCH HEAD~1
</code></pre>
<p>to get only the last commit?</p>
<p>No, that’s not right I think.</p>
<p>I know that “git clone” has a --depth option to get a limited history. </p>

**Comment by lee-dohm** [comment]
<p>It sounds like you’re using git for something that it wasn’t designed for. I suspect that you have a lot of binary files in the repository, files such as audio, image, or video files. And then on top of that, you’re intentionally not using any of the version control features of git, using it only as a remote file storage mechanism.</p>
<p>Unfortunately, it’s not possible to tell git to work in such a way. It’s going to assume that you’re trying to use it as a version control system and keep extra data around in case you decide to refer back to it. You <em>could</em> look into <a href="https://git-scm.com/docs/git-gc" rel="nofollow noopener">the <code>git gc</code> command</a> to see if it could help, but I would still recommend using a different tool that is better designed for what you’re trying to do.</p>
<p>I hope that helps!</p>

**Comment by fire-eggs** [comment]
<p>Potentially useful <a href="https://github.blog/2018-03-05-measuring-the-many-sizes-of-a-git-repository/" rel="nofollow noopener">article</a> about using git-sizer to determine how your repository is “too big”.</p>
<p>Also, is 500M “size” or “allocated size”? I have a 5M repo which occupies 75M of disk space on one device - each of many 1K files occupies 128K of allocated disk space.</p>

**Comment by hl2guide** [comment, marked as answer]
<p>Thanks everyone for the replies :smiley:</p>
<p>To prevent bloat I’ve decided to just delete the local repo folder and clone down the repo after every 20 commits.</p>
